pkg/daemon/update.go

```diff
@@ -1432,7 +1432,37 @@ func (dn *Daemon) enableUnits(units []string) error {
 	args := append([]string{"enable"}, units...)
 	stdouterr, err := exec.Command("systemctl", args...).CombinedOutput()
```

Comment:
The effect of this is that any failure to enable the units will trigger the check. Since you are changing on the on-disk state, you may have to do a `systemctl daemon-reload`. 

Under the axiom that RHEL 7 can't handle null-target symlinks, I wonder if:
- check for symlinks in `/etc/systed/system` (since you can't assume the unit is just for multi-user.target.wants)
- check if the target is dangling, remove it
- then systemctl-enable
- guard for RHEL 7 only

Reply:
> The effect of this is that any failure to enable the units will trigger the check. Since you are changing on the on-disk state, you may have to do a systemctl daemon-reload.

Right, and I propagate the old error just in case. Put another way I am just performing one extra "cleanup" when I fail the first time, so normal operation specifically doesn't hit this.

> check for symlinks in /etc/systed/system (since you can't assume the unit is just for multi-user.target.wants)

I don't want to do that because then we'd be potentially managing items **not defined in a machineconfig** which I do not want to happen

Reply:
Basically, if user error occurs, I still want to error as usual. I don't want this to actually fix incorrect unit definitions, for example, or if a user wrote a bad link with a `files` section config. I am only attempting to clean up bad links from our own code pre 4.7. I can try to make that clearer in the commit message if you think the approach makes sense

Reply:
Would you be ok if the logic was the same, but I only attempted this on RHEL7? Or if I parse the error specifically for "file exists" which is what the rhel 7 bug was? (somewhat hacky I feel)